bld/namelist_files/namelist_definition_ctsm.xml

```diff
@@ -86,6 +86,19 @@ Type of CO2 feedback.
     diagnostic = use the diagnostic value sent from the atmosphere
 </entry>
 
+<entry id="o3_ppbv" type="real" category="clm_physics"
+       group="clm_inparm" valid_values="" >
+Atmospheric O3 molar ratio (by volume) only used when o3_type==constant (nmol/mol)
+(Set by CCSM_O3_PPBV)
+</entry>
+
+<entry id="o3_type" type="char*16" category="clm_physics"
+       group="clm_inparm" valid_values="constant,diagnostic" >
+Type of O3 feedback.
+    constant   = use the input o3_ppbv value
+    disgnostic = use diagnostic values from the atmosphere
+</entry>
+
```

Comment:
In CLMBuildNamelist.pm
`
sub process_commandline {
...
o3_type               => undef,
o3_ppbv               => undef,
...
}
`
I wonder if we should initialize these here, e.g. by using "default" and 0, respectively or not.

Reply:
What do you mean here @ziu1986 ? 

> In CLMBuildNamelist.pm
> `sub process_commandline { ... o3_type => undef, o3_ppbv => undef, ... }`
> I wonder if we should initialize these here, e.g. by using "default" and 0, respectively or not.

Is the question about initializing a real and a character to the same default value `undef`? I do not understand the different initialization rules for types (characters, reals, logical) and command-line options either. 


What rules do you follow to initialize the different choices here @billsacks? It seems like namelists are initialized to 0 (like `fire_emis=0` and `ignore_warnings=0`), etc? Why are there different choices for the list `clm_demand="null"` and `glc_nec="default"`?

Reply:
Sorry, there's a lot of old code here, and I wouldn't be surprised if some of it is inconsistent. This is in need of a rewrite, which is on our to do list....

@ziu1986, it looks like the way you have done this is to add command-line flags. Maybe you were following the co2 example here. However, I think we don't want to do that for ozone, at least not for now: that adds extra complexity that is unnecessary here. It's great that you introduced a new function for this, but I think what's best is to keep this significantly simpler: Get rid of the code you added to add these as command-line flags, and instead just have a single line for each variable, like this:

```perl
add_default($opts,  $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'o3_ppbv');
```

You could still keep the check for < 0, which is nice; now, though, I think it would look like this:

```perl
      if ( $nl->get_value('o3_ppbv') < 0.0 ) {
        $log->fatal_error("o3_ppbv can NOT be less than zero.");
      }
```

I think that, for now, we should get rid of o3_type, since the only type we'll support is 'constant' - so let's just assume we're using constant o3, specified in the namelist.

By the way, I see that use_ozone was never added to CLMBuildNamelist; this was probably my fault. I think what we'll want to do here is have a more general-purpose `setup_logic_ozone` that sets o3_ppbv along with use_ozone and ozone_mode (if we go with those variable names, as discussed above). I can help with this if what I'm saying isn't clear.

Let me know if you have questions about this. Or let me know if you'd like me to just push a change that implements what I'm saying, since a few lines of code may be worth 1000 words here.